### ISSUE TITLE
Add Winding Order calculation

### ISSUE THREAD

**Issue by peterstace** [peterstace/simplefeatures#237]
We want a way to calculate the winding order of linear rings. The motivating usecase of this is as a way to detect the need to re-orient a linear ring for inclusion in a shapefile. The API design should be agnostic of that though, and applicable to more general scenarios.

A bit of thought needs to go into the best way to design the API for this. An obvious first attempt at an API design would be:

```
func IsWindingOrderClockwise(seq Sequence) bool {
    ...
}
```

This raises some questions of what to do in the following scenarios though:

- What if there aren't enough points? We need at least 3 points to reasonably define winding order. Should we return false if there aren't enough points?
- What if the input Sequence isn't actually a LinearRing? E.g. start and endpoints aren't the same, or contains self intersections.

Self intersection checks can be expensive, and may have already been performed (or the input may otherwise have been determined to be a valid linear ring).

If the function was instead:

```
func IsOuterRingWindingOrderClockwise(poly Polygon) bool {
    ...
}
```

Then we're in a better assumption to assume that the outer linear ring is actually a linear ring (since constructed polygons are assumed to have passed validity checks). But that signature doesn't fit our motivating usecase too well, since we would also want to re-orient _inner_ rings as well.

A bit more thought needs to go into the API design before we can address this.

**Comment by peterstace** [comment]
Another alternative is to go with the approach of passing in a `Sequence`, but then documenting that if the user passes in a sequence that self intersects, then the output isn't well defined.

```
// IsWindingOrderClockwise ... blah blah ... if you pass in a self interesting sequence then don't rely on the output having any particular meaning ...
func IsWindingOrderClockwise(seq Sequence) bool {
    ...
}
```

**Comment by peterstace** [comment]
We may be hitting into the [xy problem](http://xyproblem.info/) here...

The problem really is that we want to orient polygons and multipolygons as appropriate for shapefiles.

So we could just add methods to do that. E.g.

```
// ForceRHR returns the version of the Polygon that
// follows the right hand rule (i.e. the area that is bounded
// by the polygon is to the right of the boundary).
func (p Polygon) ForceRHR() Polygon {
    ....
}
```

**Comment by peterstace** [comment]
Fixed via #240
